### PR TITLE
Polish interfaces in OptimalBranchingCore

### PR DESCRIPTION
**Pull request by GiggleLiu**
```julia
julia> using OptimalBranchingCore

julia> tbl = BranchingTable(5, [
        [[0, 0, 0, 0, 1], [0, 0, 0, 1, 0]],
        [[0, 0, 1, 0, 1]],
        [[0, 1, 0, 1, 0]],
        [[1, 1, 1, 0, 0]]])
BranchingTable{BitBasis.LongLongUInt{1}}
10000, 01000
10100
01010
00111

julia> candidates = collect(OptimalBranchingCore.candidate_clauses(tbl))
14-element Vector{Clause{BitBasis.LongLongUInt{1}}}:
 Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ ¬#2 ∧ ¬#3 ∧ ¬#4 ∧ #5
 Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ ¬#3 ∧ #4 ∧ ¬#5
 Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ ¬#2 ∧ #3 ∧ ¬#4 ∧ #5
 Clause{BitBasis.LongLongUInt{1}}: #1 ∧ #2 ∧ #3 ∧ ¬#4 ∧ ¬#5
 Clause{BitBasis.LongLongUInt{1}}: #2 ∧ ¬#5
 Clause{BitBasis.LongLongUInt{1}}: #3 ∧ ¬#4
 Clause{BitBasis.LongLongUInt{1}}: ¬#1
 Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ ¬#2 ∧ ¬#4 ∧ #5
 Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ ¬#2 ∧ ¬#3 ∧ #4 ∧ ¬#5
 Clause{BitBasis.LongLongUInt{1}}: ¬#5
 Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ ¬#2
 Clause{BitBasis.LongLongUInt{1}}: ¬#4
 Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ ¬#3
 Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ #2 ∧ ¬#3 ∧ #4 ∧ ¬#5

julia> Δρ = [length(literals(sc)) for sc in candidates]
14-element Vector{Int64}:
 5
 4
 5
 5
 2
 2
 1
 4
 5
 1
 2
 1
 2
 5

julia> res_ip = OptimalBranchingCore.minimize_γ(tbl, candidates, Δρ, IPSolver())
OptimalBranchingResult{BitBasis.LongLongUInt{1}, Int64}([4, 2, 3], DNF{BitBasis.LongLongUInt{1}}(Clause{BitBasis.LongLongUInt{1}}[Clause{BitBasis.LongLongUInt{1}}: #1 ∧ #2 ∧ #3 ∧ ¬#4 ∧ ¬#5, Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ ¬#3 ∧ #4 ∧ ¬#5, Clause{BitBasis.LongLongUInt{1}}: ¬#1 ∧ ¬#2 ∧ #3 ∧ ¬#4 ∧ #5]), [5, 4, 5], 1.2671683045421243)
```

### CODE REVIEW

**Comment by GiggleLiu** [comment]
This PR is used to support the appenx. The motivation is to make the results more readable.

File: lib/OptimalBranchingCore/src/setcovering.jl [hunk 1]
```diff
@@ -92,64 +92,136 @@ function bisect_solve(f, a, fa, b, fb)
...
+
+The result type for the optimal branching rule.
+
+### Fields
+- `selected_ids::Vector{Int}`: The indices of the selected rows in the branching table.
+- `optimal_rule::DNF{INT}`: The optimal branching rule.
+- `branching_vector::Vector{T<:Real}`: The branching vector that records the size reduction in each subproblem.
+- `γ::Float64`: The optimal γ value (the complexity of the branching rule).
+"""
+struct OptimalBranchingResult{INT <: Integer, T <: Real}
```
Comment (ArrogantGao) [inline lib/OptimalBranchingCore/src/setcovering.jl:105]: I suggest here we may add a field `variables`, which represents the varibales being branched.

**Comment by ArrogantGao** [comment]
for the example to be used in the article, I suggest something like this: https://github.com/ArrogantGao/OptimalBranching_benchmark/tree/main/rule_discovery

File: lib/OptimalBranchingCore/src/setcovering.jl (hunk 1 above)
Comment (GiggleLiu) [reply lib/OptimalBranchingCore/src/setcovering.jl:105]: I do not think it is possible, we do not have the variables context in the input of `minimum_γ`.

Comment (ArrogantGao) [reply lib/OptimalBranchingCore/src/setcovering.jl:105]: I see.
Then I have no doubts about this PR, I will merge it to main.

**Pull request merged**